openmetadata-ui/src/main/resources/ui/src/pages/MlModelPage/MlModelPage.component.tsx

```diff
@@ -0,0 +1,256 @@
+      });
+  };
+
+  const saveUpdatedMlModelData = (
+    updatedData: Mlmodel
+  ): Promise<AxiosResponse> => {
+    const jsonPatch = compare(mlModelDetail, updatedData);
+
+    return patchMlModelDetails(
+      mlModelDetail.id,
+      jsonPatch
+    ) as unknown as Promise<AxiosResponse>;
+  };
+
+  const descriptionUpdateHandler = (updatedMlModel: Mlmodel) => {
+    saveUpdatedMlModelData(updatedMlModel)
+      .then((res: AxiosResponse) => {
+        if (res.data) {
+          const { description } = res.data;
+          setMlModelDetail((preVDetail) => ({
+            ...preVDetail,
+            description: description,
+          }));
+        } else {
+          throw jsonData['api-error-messages']['update-description-error'];
+        }
+      })
+      .catch((err: AxiosError) => {
+        showErrorToast(
+          err,
+          jsonData['api-error-messages']['update-description-error']
+        );
+      });
+  };
+
+  const followMlModel = () => {
+    addFollower(mlModelDetail.id, USERId)
+      .then((res: AxiosResponse) => {
+        if (res.data) {
+          const { newValue } = res.data.changeDescription.fieldsAdded[0];
+
+          setMlModelDetail((preVDetail) => ({
+            ...preVDetail,
+            followers: [...(mlModelDetail.followers || []), ...newValue],
+          }));
+        } else {
+          throw jsonData['api-error-messages']['update-entity-follow-error'];
+        }
+      })
+      .catch((err: AxiosError) => {
+        showErrorToast(
+          err,
+          jsonData['api-error-messages']['update-entity-follow-error']
+        );
+      });
+  };
+
+  const unfollowMlModel = () => {
+    removeFollower(mlModelDetail.id, USERId)
+      .then((res: AxiosResponse) => {
+        if (res.data) {
+          const { oldValue } = res.data.changeDescription.fieldsDeleted[0];
+          setMlModelDetail((preVDetail) => ({
+            ...preVDetail,
+            followers: (mlModelDetail.followers || []).filter(
+              (follower) => follower.id !== oldValue[0].id
+            ),
+          }));
+        } else {
+          showErrorToast(
+            jsonData['api-error-messages']['update-entity-unfollow-error']
+          );
+        }
+      })
+      .catch((err: AxiosError) => {
+        showErrorToast(
+          err,
+          jsonData['api-error-messages']['update-entity-unfollow-error']
+        );
+      });
+  };
+
+  const onTagUpdate = (updatedMlModel: Mlmodel) => {
+    saveUpdatedMlModelData(updatedMlModel)
+      .then((res: AxiosResponse) => {
+        if (res.data) {
+          setMlModelDetail((preVDetail) => ({
+            ...preVDetail,
+            tags: res.data.tags,
+          }));
+        } else {
+          throw jsonData['api-error-messages']['update-tags-error'];
+        }
+      })
+      .catch((err: AxiosError) => {
+        showErrorToast(
+          err,
+          jsonData['api-error-messages']['update-tags-error']
+        );
+      });
+  };
```

Comment:
can we create util function something like this and use this across all API's? 
```
checkAPIError ( data:[any] , errorMessage:string){
      if(data.data > 0){
          return
      }
      else
        throw jsonData['api-error-messages'][errorMessage]
  }
```

we can use it like this

```
try{
const res = await   saveUpdatedMlModelData(updatedMlModel);
checkApiError(res,"string");
 setMlModelDetail((preVDetail) => ({
            ...preVDetail,
            mlFeatures: res.data.mlFeatures,
          }));
}
catch(err){
 showErrorToast(err);
}
```
@shahsank3t @vivekratnavel what your point on this ? any suggestion ?

Reply:
Yes, we can add util and start using it everywhere.

Reply:
Thanks, @yug-shah0106 , I will make the required changes.